website/blog/metric_labels.md

```diff
@@ -0,0 +1,130 @@
+    attribute.Key("status_code").Int(404),
+))
+```
+
+## Drawbacks
+
+Specifying labels as a set of key-value pairs like this has a number of
+disadvantages.
+
+- **Misspelling.** You can misspell label names. For example, you might
+  accidentally use label name `"stats_code"` instead of `"status_code"`.
+- **Mistyping.** You can use label values with the wrong type. For example, you
+  might accidentally pass a status like `"404 Not Found"` instead of the status
+  code `404`.
+- **Misremembering.** You can misremember the name of a label completely. Was it
+  `"path"` or `"endpoint"` or `"URL"`? You can accidentally use the wrong name.
+- **Too many labels.** You can pass a label that wasn't defined on the metric.
+  If we remove the `status_code` label, for example, we have to be careful to
+  update our code everywhere that uses it.
+- **Too few labels.** You can forget to pass a label. If we add a new label to a
+  metric, for example, we again have to carefully update our code to use it.
+
+## Service Weaver's Approach
+
+Service Weaver avoids these drawbacks by integrating metric labels with Go's
+type system. Specifically, Service Weaver represents metric labels as structs
+and uses generics to instantiate metrics. Here's how to declare the
+`http_requests` metric using [Service Weaver's API][weaver_api]:
+
+```go
+type labels struct {
+    Path       string `weaver:"path"`
+    StatusCode int    `weaver:"status_code"`
+}
+
+var requests = metrics.NewCounterMap[labels](
+    "http_requests",
+    "Number of HTTP requests.",
+)
+```
+
+And here's how to increment the counter:
+
+```go
+requests.Get(labels{Path: "/foo", StatusCode: 404}).Inc()
+```
+
+Labels can be any struct where every field is a string, bool, or integer.
+`NewCounterMap` panics if you call it with an invalid label struct. By
+leveraging Go's type system, Service Weaver's API avoids the disadvantages of
+key-value labels described above.
+
+- **Misspelling.** If you misspell a field name, your code will not compile
+  (e.g., `labels{Path: "/foo", StatsCode: 404}`).
+- **Mistyping.** If you pass a label value of the wrong type, your code will not
+  compile (e.g., `labels{Path: "/foo", StatusCode: "404 Not Found"}`).
+- **Misremembering.** If you use the wrong field name, your code will not
+  compile (e.g., `labels{Endpoint: "/foo", StatusCode: 404}`).
+- **Too many labels.** If you pass a label that does not exist, your code will
+  not compile (e.g., `labels{Path: "/foo", StatusCode: 404, ContentLength: 42}`).
+- **Too few labels.** This one is a bit tricky because Go does not require you
+  to initialize every field in a struct. However, if you omit field names in a
+  struct literal, you must provide every field or your code will not compile
+  (e.g., `labels{"/foo"}`).
+
+These strongly typed metric labels are just one of the ways that Service Weaver
+is making it easier to write distributed systems. Read [the Service Weaver
```

Comment:
Maybe we should briefly mention here, that we support various types of metrics like counters, gauges and histograms, so we can cover things that other systems do too.